api-server/scanner-lib/src/blockchain_state/mod.rs

```diff
@@ -1241,12 +1283,14 @@ async fn update_tables_from_transaction_inputs<T: ApiServerStorageWrite>(
                         order.fill(&chain_config, block_height, *fill_amount_in_ask_currency);
 
                     db_tx.set_order_at_height(*order_id, &order, block_height).await?;
+                    update_tokens_in_transaction(&order, &mut transaction_tokens);
                 }
                 OrderAccountCommand::ConcludeOrder(order_id) => {
                     let order = db_tx.get_order(*order_id).await?.expect("must exist");
                     let order = order.conclude();
 
                     db_tx.set_order_at_height(*order_id, &order, block_height).await?;
+                    update_tokens_in_transaction(&order, &mut transaction_tokens);
                 }
                 OrderAccountCommand::FreezeOrder(order_id) => {
                     let order = db_tx.get_order(*order_id).await?.expect("must exist");
```

Comment:
I wonder if order freeze should be considered a "token" transaction as well, if the order involves tokens.

I mean, you already count ConcludeOrder unconditionally, even if it doesn't involve any token movement.